xclim/indices/_conversion.py

```diff
@@ -585,18 +585,74 @@ def snowfall_approximation(
 
     Notes
     -----
-    The following methods are available to approximate snowfall:
+    The following methods are available to approximate snowfall and are drawn from the
+    Canadian Land Surface Scheme (CLASS, [Verseghy09]_).
 
-    - "binary" : When the given temperature is under a given threshold, precipitation
+    - "binary" : When the temperature is under the freezing threshold, precipitation
         is assumed to be solid. The method is agnostic to the type of temperature used
         (mean, maximum or minimum).
+    - "brown" : The phase between the freezing threshold goes from solid to liquid linearly
+        over a range of 2°C over the freezing point.
+    - "auer" : The phase between the freezing threshold goes from solid to liquid as a degree six
+        polynomial over a range of 6°C over the freezing point.
 
+    .. [Verseghy09]: Diana Verseghy (2009), CLASS – The Canadian Land Surface Scheme (Version 3.4), Technical
+    Documentation (Version 1.1), Environment Canada, Climate Research Division, Science and Technology Branch.
     """
-    thresh = convert_units_to(thresh, tas)
+    # https://gitlab.com/cccma/classic/-/blob/master/src/atmosphericVarsCalc.f90
+
     if method == "binary":
-        prsn = pr.where(tas < thresh, 0)
+        thresh = convert_units_to(thresh, tas)
+        prsn = pr.where(tas <= thresh, 0)
+
+    elif method == "brown":
+        # Freezing point + 2C in the native units
+        upper = convert_units_to(convert_units_to(thresh, "degC") + 2, tas)
+        thresh = convert_units_to(thresh, tas)
+
```

Comment:
```suggestion
        thresh = convert_units_to(thresh, tas)
        upper = convert_units_to("2 degC", tas) + thresh
```
One less `convert_units_to`  calls!

Reply:
Hum, non, si tu transformes 2C en Kelvin, tu vas te retrouver avec upper ~ 500 K

Reply:
Woups. C'est bien vrai tout ça, :upside_down_face: .